File: 프로그래머스/Level.1/8. 부족한 금액 계산하기.js
function solution(price, money, count) {
  const arr = [];

  while (0 < count) {
    arr.push(price * count);
    count--;
  }

  const res = arr.reduce((acc, cur) => acc + cur, 0);

  return res - money < 0 ? 0 : res - money;
}

solution(3, 20, 4);

///////다른 풀이

function solution(price, money, count) {
  let answer = 0;

  for (let i = 1; i <= count; i++) {
    answer += price * i;
  }

  return answer > money ? answer - money : 0;
}

File: 프로그래머스/Level.1/8. 부족한 금액 계산하기.py
def solution(price, money, count):
    # 총 놀이기구 이용금액 변수선언
    total = 0
    # count 범위만큼 반복문
    for i in range(1, count+1):
        # price * count
        total += price * i
    # 금액이 부족하지 않으면 0리턴
    return 0 if total - money < 0 else total - money


# 팀 코드
def solution(price, money, count):
    total = sum(price * i for i in range(1, count + 1)) 
    # 1부터 count 번 반복해서 price를 곱한 값들의 총합은 total에 담아줌
    return 0 if money - total >= 0 else total - money
    # 가진돈이 모자라지 않으면 0 그 외에는 총합에서 가진돈을 빼줌  

solution(3, 20, 4)